### PR TITLE
Update docs and add doc for seting up monorepo

### PR DESCRIPTION
**Pull request by mingyu89**
Fix docs after project layout refactoring.

If mono repo is used, current step is to use mlops-stacks to generate another project and copy a bunch of files. It might be better to include a cookiecutter project inside the generated project and directly generated these files.

In current approach, files in .github/workflows/ could have name conflicts. We are asking user to rename files before copying to the mono repo. Another option is to add the project name as prefix to these yaml files.

Anyway, this PR is to fix docs. For mono repo support, we can decide and do more later.

### CODE REVIEW

**Comment by mingyu89** [comment]
Feel free to comment as much as possible. 
Generated project https://github.com/mingyu89/test-repo1/tree/update-doc

**Comment by mingyu89** [comment]
Ready for an initial review

File: {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/README.md [hunk 1]
```diff
@@ -25,10 +25,27 @@ After that, follow the [ML pull request guide](docs/ml-pull-request.md)
...
+| Data Scientist                | Get started writing ML code for a brand new project                           | {% if cookiecutter.include_feature_store == "yes" %}[ML quickstart](docs/ml-developer-guide-fs.md).{% else %}[ML quickstart](docs/ml-developer-guide.md){% endif %} |
+| Data Scientist                | Update production ML code (e.g. model training logic) for an existing project | [ML pull request guide](docs/ml-pull-request.md)                                                                                                                    |
+| Data Scientist                | Modify production model ML resources, e.g. model training or inference jobs   | [ML resource config guide]({{cookiecutter.project_name}}/terraform/README.md)                                                                                       |
+| MLOps / DevOps                | Set up CI/CD and ML pipeline resource deployment for the current ML project   | [MLOps setup guide](docs/mlops-setup.md)                                                                                                                            |
+
+## Mono repo
```
Comment (vladimirk-db) [inline {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/README.md:36]:
```suggestion
## Monorepo
```

File: {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/README.md [hunk 2]
```diff
@@ -25,10 +25,27 @@ After that, follow the [ML pull request guide](docs/ml-pull-request.md)
...
+
+It's possible to use the repo as a mono repo that contains multiple projects. All projects share the same workspace setup and service principal.
```
Comment (vladimirk-db) [inline {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/README.md:38]:
```suggestion
It's possible to use the repo as a monorepo that contains multiple projects. All projects share the same workspace setup and service principal.
```

File: {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/README.md [hunk 3]
```diff
@@ -25,10 +25,27 @@ After that, follow the [ML pull request guide](docs/ml-pull-request.md)
...
+
+For example, assuming there's existing repo with root directory name `monorepo_root_dir` and project name `project1`
+1. Create another project from cookiecutter with project name `project2` and root direction name `project2`.
```
Comment (vladimirk-db) [inline {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/README.md:41]:
```suggestion
1. Create another project from cookiecutter with project name `project2` and root directory name `project2`.
```

File: {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/README.md [hunk 4]
```diff
@@ -25,10 +25,27 @@ After that, follow the [ML pull request guide](docs/ml-pull-request.md)
...
+2. Copy the internal directory `project2/project2` to root directory of existing repo `monorepo_root_dir/project2`.
+{% if cookiecutter.cicd_platform in ["gitHub", "gitHubEnterprise"] -%}
+3. Rename yaml files in  `project2/.github/workflows/` so that there won't be name conflicts.
+4. Copy yaml files from `project2/.github/workflows/` to `monorepo_root_dir/.github/workflows/`
+{% endif -%}
+{%- if cookiecutter.cicd_platform == "azureDevOpsServices" %}
+3. Rename yaml files in  `project2/.azure/devlops-pipelines/` so that there won't be name conflicts.
+4. Copy yaml files from `project2/.azure/devlops-pipelines/` to `monorepo_root_dir/.azure/devlops-pipelines/`
+{% endif -%}
```
Comment (vladimirk-db) [inline {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/README.md:50]: For mono-repos, we could generate project-name suffixes to the GH workflows files. E.g., run-test-project1.yaml

Comment (vladimirk-db) [reply {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/README.md:50]: Not urgent or blocking for this PR.

File: {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/docs/project-overview.md [hunk 5]
```diff
@@ -24,48 +24,75 @@ This project contains the following components:
...
-│
-├── requirements.txt   <- Specifies Python dependencies for ML code (model training, batch inference, etc).
-│
-├── tests              <- Unit tests for the modules under `features`.
+{{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}         <- Root directory. Both mono repo and poly repo are supported.
 │
+├── {{cookiecutter.project_name}}       <- Contains python code, notebooks and ML resources related to one ML project. 
+│   │
+{% if cookiecutter.include_feature_store == "yes" -%}
+│   ├── training                <- Training folder contains Notebook that trains and regisetered the model with feature store support.
```
Comment (vladimirk-db) [inline {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/docs/project-overview.md:32]:
```suggestion
│   ├── training                <- Training folder contains Notebook that trains and registers the model with feature store support.
```

File: {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/docs/project-overview.md [hunk 6]
```diff
@@ -24,48 +24,75 @@ This project contains the following components:
...
+│   │
+│   ├── feature_engineering     <- Feature computation code (Python modules) that implements the feature transforms.
+│   │                              The output of these transforms get persisted as Feature Store tables. Most development
+│   │                              work happens here.
+│   │
+│   ├── validation              <- Optional model validation step before deploying a model.
+│   │
+│   ├── monitoring              <- Model monitoring, feature monitoring, etc.
+│   │
+│   ├── deployment              <- Model deployment and endpoint deployment.
```
Comment (vladimirk-db) [inline {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/docs/project-overview.md:42]: Rephrase as Deployment and Batch inference workflows?

File: {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/docs/project-overview.md [hunk 7]
```diff
@@ -24,48 +24,75 @@ This project contains the following components:
...
+│   │   │
+│   │   ├── batch_inference     <- Batch inference code that will run as part of schduled workflow.
```
Comment (vladimirk-db) [inline {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/docs/project-overview.md:44]: Please fix typos.

File: {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/docs/project-overview.md [hunk 8]
```diff
@@ -24,48 +24,75 @@ This project contains the following components:
...
-│── recipe.yaml      <- The main recipe configuration file that declaratively defines the attributes and behavior
-│                         of each recipe step, such as the input dataset to use for training a model or the
-│                         performance criteria for promoting a model to production.
-│
-├── profiles           <- Environment-specific (e.g. dev vs test vs prod) configurations for MLflow pipeline execution.
-│
-├── requirements.txt   <- Specifies Python dependencies for ML code (for example: model training, batch inference).
-│
-├── tests              <- Unit tests for the modules under `steps`.
+│   ├── training                <- Folder for model development via MLflow recipe.
```
Comment (vladimirk-db) [inline {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/docs/project-overview.md:51]: MLflow recipes.

File: {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/docs/project-overview.md (hunk 7 above)
Comment (vladimirk-db) [reply {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/docs/project-overview.md:44]: Please do all these same changes below, seems like it's duplicated (does it have to be?)

File: {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/{{cookiecutter.project_name}}/deployment/batch_inference/README.md [hunk 9]
```diff
@@ -0,0 +1,2 @@
+# Batch Inference
+To set up batch inference job via scheduled databricks workflow, please refer to [{{cookiecutter.project_name}}/terraform/README.md](../../terraform/README.md#setting-up-batch-inference-job)
```
Comment (vladimirk-db) [inline {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/{{cookiecutter.project_name}}/deployment/batch_inference/README.md:2]: nit: Databricks

**Review by zhe-db — APPROVED** [review]
LGTM Thanks @mingyu89

File: {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/README.md (hunk 1 above)
Comment (mingyu89) [reply {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/README.md:36]: Thank you. Updated all `Mono repo` => `Monorepo` and `poly repo` => `polyrepo.`

File: {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/README.md (hunk 4 above)
Comment (mingyu89) [reply {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/README.md:50]: Sounds good!

File: {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/docs/project-overview.md (hunk 7 above)
Comment (mingyu89) [reply {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/docs/project-overview.md:44]: The `test` folder below it has different text. That's why I have to duplicate `deployment` in both branches.

**Comment by mingyu89** [comment]
@vladimirk-db Thank you! I have so many typos!

**Review by vladimirk-db — APPROVED** [review]
LGTM but please address comments before submitting.

**Pull request merged**